app/src/main/java/org/kiwix/kiwixmobile/nav/destination/library/OnlineLibraryFragment.kt

```diff
@@ -154,12 +154,7 @@ class OnlineLibraryFragment : BaseFragment(), FragmentActivityExtensions {
     zimManageViewModel.networkStates.observe(viewLifecycleOwner, Observer(::onNetworkStateChange))
     zimManageViewModel.shouldShowWifiOnlyDialog.observe(viewLifecycleOwner, Observer {
       if (it) {
-        dialogShower.show(
-          WifiOnly,
-          { sharedPreferenceUtil.putPrefWifiOnly(false) },
-          { onRefreshStateChange(false) }
-        )
-        zimManageViewModel.shouldShowWifiOnlyDialog.value = false
```

Comment:
I investigate a little and here is the bug, we are setting the `shouldShowWifiOnlyDialog` with nothing. We should remove this line to avoid this bug.

Reply:
I have moved this condition on  yes button action.